anu_lms.module

```diff
@@ -56,3 +57,40 @@ function anu_lms_node_delete(NodeInterface $node) {
     }
   }
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Add an option to enable locking.
+ */
+function anu_lms_form_views_form_sort_courses_sort_page_1_alter(&$form, FormStateInterface $form_state) {
+  $termId = Drupal::routeMatch()->getParameter('taxonomy_term');
+  $term = \Drupal::entityTypeManager()->getStorage('taxonomy_term')->load($termId);
+  $form['sequence'] = [
+    '#type' => 'checkbox',
+    '#title' => t('Courses must be completed in this order'),
+    '#default_value' => $term->field_enable_course_sequence->value,
+  ];
+  $form['#submit'][] = 'anu_lms_form_views_form_sort_courses_sort_page_1_submit';
+}
+
+/**
+ * Custom submit to enable locking.
+ */
+function anu_lms_form_views_form_sort_courses_sort_page_1_submit(&$form, FormStateInterface $form_state) {
+  if ($form_state->getValue('sequence')) {
+    // Enable linear progress for all courses involved.
+    foreach ($form['field_weight'] as $element) {
+      if (isset($element['entity']['#value'])) {
+        $course = $element['entity']['#value'];
+        $course->field_course_linear_progress->value = TRUE;
```

Comment:
@rodrigoaguilera 

> I'm not sure I follow your idea in the second part of the comment. You mean to disable the checkbox widget to change the progress setting on a course when it belongs to a category with lock enabled, make it checked(still not editable) and add a description that explains why?
If that is it I think is a great enhancement

That's what I understand too, and it would make sense

Reply:
Yes, that's exactly what I meant and I see that you already implemented it perfectly well :)